### PR TITLE
style(ci): lint and standardize the actions structure

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

When making changes that affected multiple actions, not having an standardized file made global replacements difficult, and sometimes it left unwanted code unchanged.

### Designs

- Make all the actions standard in structure and indentation

## Solution

- Refactor and lint the files to comply with https://json.schemastore.org/github-workflow.json

## Review

@dconnolly can also review this

### CODE REVIEW

**Review by teor2345 — COMMENTED** [review]
I have some questions about this PR:
* How can we be sure we've made all these changes consistently? Did you use search and replace, or make each change manually?
* Is there an automated linter for the formatting rules you're using, or an automated formatter to apply the rules?

This PR changes job conditions and job names, which has previously caused CI to fail or skip tests. So I've marked it as "do not merge", until we decide how risky it is. (We're currently working on some urgent NU5 and lightwalletd changes.)

**Comment by gustavovalverde** [comment]
I used https://marketplace.visualstudio.com/items?itemName=cschleiden.vscode-github-actions to ensure the conditions were kept as we expected, and https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml to lint based on GitHub Action's json schema https://json.schemastore.org/github-workflow.json (mostly automatic)

The **manual** changes requested to comply were:
- Using `macos-latest` as `macOS-latest` does not exists in the schema
- Remove `if` conditions that were not evaluating as there was no context to evaluate from
-  Use required and missing `keys` from the `inputs` in `workflow_dispatch`

Changes after revision:
- Revert to single quotes (fewer diff)
- Lint Mergify and Dependabot conf

**Review by teor2345 — COMMENTED** [review]
This seems like a potentially risky fix, that's isn't needed right now.

So I'd like to wait until after the high-priority NU5 updates are merged, and the NU5 testnet reactivation release is tagged.

**Comment by gustavovalverde** [comment]
@teor2345 @dconnolly I'd like to prioritize this PR, as it will help me identify common errors in other PRs, including parsing errors that won't be correctly parsed by GH (I added an action to lint other actions and verify shell errors)

File: .github/workflows/lint.yml
Comment (teor2345) [inline .github/workflows/lint.yml:90]: Possible bug: this action only runs when the lint workflow file changes, but we want it to run when any workflow file changes.

Or possibly any file in `.github` ?

**Comment by teor2345** [comment]
> @teor2345 @dconnolly I'd like to prioritize this PR, as it will help me identify common errors in other PRs, including parsing errors that won't be correctly parsed by GH (I added an action to lint other actions and verify shell errors)

That makes sense. Did you need it to go in before we tag the NU5 testnet rollback version?

(We're currently waiting for the `zcashd` team to release 4.7.0 and pick an activation height. Hopefully that will happen some time in the next few days.)

Here's what I think this PR does:

No behaviour change:
- use standard formatting

Behaviour change:
- change when some workflows run
- add a job that lints workflows

Unsure:
- remove some job conditions - only a behaviour change if the conditions were skipping the job

Do any of these changes need to merge sooner?

If we need some changes sooner, we could reduce the risk by splitting it into multiple PRs. Or by testing a Rust PR immediately after it merges, and reverting if it causes problems.

File: .github/workflows/lint.yml
Comment (dconnolly) [reply .github/workflows/lint.yml:90]: I think only files in `.github/workflows/`

File: .github/workflows/lint.patch.yml
Comment (dconnolly) [inline .github/workflows/lint.patch.yml:7]: Only PR? Just confirming

File: .github/workflows/lint.yml
Comment (dconnolly) [inline .github/workflows/lint.yml:90]: 🎉

**Review by dconnolly — CHANGES_REQUESTED** [review]
Can we make sure `actionlint` runs at least any time any `.github/workflows/*` file is changed? I'm so happy to have a GH workflow linter!

**Review by dconnolly — DISMISSED** [review]
Can we make sure `actionlint` runs at least any time any `.github/workflows/*` file is changed? I'm so happy to have a GH workflow linter!

File: .github/workflows/lint.yml
Comment (teor2345) [reply .github/workflows/lint.yml:90]: @gustavovalverde are the mergify and dependabot changes in this PR from the linter?
Or are they manual changes?

Comment (gustavovalverde) [reply .github/workflows/lint.yml:90]: @teor2345 yes, all the changes are from the linter. Except the `macos-latest` change, as that was required to be manually changed as this is the actual name used by GHA and required to be changed by the linter

File: .github/workflows/lint.patch.yml
Comment (gustavovalverde) [reply .github/workflows/lint.patch.yml:7]: Yes, there was no value added on using it on push, and the `reviewdog/action-actionlint@v1.21.0` required the triggering event to be a PR, and making another workflow just for this was too cumbersome (until we find a better approach).

Comment (gustavovalverde) [reply .github/workflows/lint.patch.yml:7]: Yes

**Comment by gustavovalverde** [comment]
@teor2345 answering you here:

> Here's what I think this PR does:
> 
> No behaviour change:
> 
> use standard formatting
> Behaviour change:
> 
> change when some workflows run
> add a job that lints workflows
> Unsure:
> 
> remove some job conditions - only a behaviour change if the conditions were skipping the job

**Change when some workflow run:**
- This was the changed condition https://github.com/ZcashFoundation/zebra/pull/3940/files#diff-107e910e9f2ebfb9a741fa10b2aa7100cc1fc4f5f3aca2dfe78b905cbd73c0d2 which I recently explained here: https://github.com/ZcashFoundation/zebra/pull/3940#discussion_r841316337 and it should be "safe" as it's just running the same linting but just on PR instead of push, and adding the workflow linters. This won't consume much time on each PR until we have a better solution.

**Add a job that lints workflows:**
- As this jobs just raises warnings and it doesn't try to lint the files automatically, it should be safe enough.

**Remove some job conditions**
- This conditions were wrongly added to a patch file which didn't have inputs https://github.com/ZcashFoundation/zebra/pull/3940/files#diff-74d6f532ff664f2d9c2ee8b539caf7545179c52ae354f21a6b6bdbb6b2a0d084

I reduced some diffs so we're basically just adding the formatting and the linting behavior change.

File: .github/workflows/lint.yml
Comment (gustavovalverde) [reply .github/workflows/lint.yml:90]: Fixed here: https://github.com/ZcashFoundation/zebra/pull/3940/commits/22543b71c8e42763d0ffd24022b3f0c6afa9ff11

**Review by teor2345 — COMMENTED** [review]
It looks like there might be some bugs in this PR:
- the full validation sync fails with the wrong image name
- shellcheck complains about quoting (this could be the cause of the sync name bug)
- the linter complains that `checkpoint_sync` is not defined in the job inputs

It looks like the `zcashd` team are still working on their 4.7.0 release. So we probably have time to merge this PR, fix any bugs, then test and merge the NU5 release.

**Comment by gustavovalverde** [comment]
@teor2345 this is ready. The shellcheck can be validated later, those shell commands have been like that for a while, I don't think those recommendations are completely valid for the specific use case.

I fixed the `checkpoint_sync` issue as that one was needed.

**Comment by teor2345** [comment]
> @teor2345 this is ready. The shellcheck can be validated later, those shell commands have been like that for a while, I don't think those recommendations are completely valid for the specific use case.

I checked each shellcheck warning.

Some could be fixed, but none of them seem to be causing problems right now. They will be ok as long as the names don't contain special characters. (And we control those names.)

Some of them might need to be ignored.

**Review by teor2345 — APPROVED** [review]
Looks good, thanks for your patience with this!

**Comment by teor2345** [comment]
Merge queue: sync past mandatory checkpoint:
> Error: No such container: klt-sync-checkpoint-4078-merge-97382fe-qxsz

https://github.com/ZcashFoundation/zebra/runs/5963991145?check_suite_focus=true#step:9:115

**Comment by teor2345** [comment]
@mergifyio update

**Review by teor2345 — COMMENTED** [review]
The "Test full validation sync from cached state" job logs this warning:
>  Warning: The "create_credentials_file" option is true, but the current GitHub workspace is empty. Did you forget to use "actions/checkout" before this step? If you do not intend to share authentication with future steps in this job, set "create_credentials_file" to false.

https://github.com/ZcashFoundation/zebra/runs/5964749623?check_suite_focus=true#step:4:71

**Comment by gustavovalverde** [comment]
Yeah, I have that warning pending. I don't want to do checkout if it's not needed, but this is working without using checkout, but I'm not sure it will still work if I remove the `create_credentials_file`. I'll test that in the documentation PR

**Review by teor2345 — CHANGES_REQUESTED** [review]
I'm still seeing these container name errors:

> the full validation sync fails with the wrong image name

> Merge queue: sync past mandatory checkpoint:
> 
> > Error: No such container: klt-sync-checkpoint-4078-merge-97382fe-qxsz
> 
> https://github.com/ZcashFoundation/zebra/runs/5963991145?check_suite_focus=true#step:9:115

**Comment by gustavovalverde** [comment]
This is, once again, the error caused by partitions in GCP 
image

This happened before in other PRs, I'll follow up with the ticket. But it's not related to the name of the container.

I'll also create a PR to show this kind of issues when the container does not start, so it's easier for the team identifying the root cause without having to get into GCP and search for the logs

**Review by teor2345 — APPROVED** [review]
Thanks, looks good!

**Pull request merged**
